DataStructures.Tests/BinaryTreeTests.cs

```diff
@@ -0,0 +1,127 @@
+﻿using DataStructures.Trees;
+using System.Text;
+using Xunit;
+
+namespace DataStructures.Tests
+{
+    public class TreeTests
+    {
+        [Fact]
+        public void Add_value_to_empty_tree()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+
+            // Act
+            tree.Add(4);
+
+            //Assert
+            Assert.Equal(4, tree.Root.Value);
+        }
+
+        [Fact]
+        public void Add_one_Left_and_one_Right_node()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+
+            // Act
+            tree.Add(8);
+            tree.Add(4);
+            tree.Add(5);
+
+            //Assert
+            Assert.Equal(8, tree.Root.Value);
+            Assert.Equal(4, tree.Root.Left.Value);
+            Assert.Equal(5, tree.Root.Right.Value);
+        }
+
+        [Fact]
+        public void Contains_a_value_to_find_Node()
+        {
+            // Arrange
+            BinarySearchTree tree = new BinarySearchTree();
+            tree.Add(8);
+            tree.Add(4);
+            tree.Add(5);
+            tree.Add(1);
+            tree.Add(6);
+            tree.Add(3);
+
+            // Act
+            bool trueResult = tree.Contains(tree.Root, 8);
+            bool falseResult = tree.Contains(tree.Root, 15);
+
+            //Assert
+            Assert.True(trueResult);
+            Assert.False(falseResult);
```

Comment:
It might be a good idea to add a few more asserts here just to verify that your contains is actually traversing through the tree to find values that are deeper like so:
[Fact]
        public void Contains_a_value_to_find_Node()
        {
            // Arrange
            BinarySearchTree tree = new BinarySearchTree();
            tree.Add(8);
            tree.Add(4);
            tree.Add(5);
            tree.Add(1);
            tree.Add(6);
            tree.Add(3);
            tree.Add(10);

            // Act
            bool trueResult = tree.Contains(tree.Root, 8);
            bool falseResult = tree.Contains(tree.Root, 15);

            //Assert
            Assert.True(trueResult);
            Assert.False(falseResult);
            Assert.True(tree.Contains(tree.Root, 6));
            Assert.True(tree.Contains(tree.Root, 3));
            Assert.True(tree.Contains(tree.Root, 6));
            Assert.True(tree.Contains(tree.Root, 10));
        }

Reply:
P.S. I did so and it still passed, but as for an approach to testing you want to verify that you are checking different outcomes for different inputs.